src/services/userService.js

```diff
@@ -0,0 +1,51 @@
+let users = [];
+let userIdCounter = 1;
+
+const resetUsers = () => {
+  users = [];
+  userIdCounter = 1;
+};
+
+const getAllUsers = () => {
+  return users;
+};
+
+const getUserById = (id) => {
+  return users.find((person) => person.id === +id) || null;
+};
+
+const createUser = (name) => {
+  const user = {
+    id: userIdCounter++,
+    name,
+  };
+
+  users.push(user);
+
+  return user;
+};
+
+const updateUser = ({ id, name }) => {
+  const user = getUserById(+id);
+
+  if (!user) {
+    return null;
+  }
+
+  Object.assign(user, { name });
+
+  return user;
+};
```

Comment:
The updateUser function could be improved by adding error handling. For example, you should check if the name is not empty before updating the user.